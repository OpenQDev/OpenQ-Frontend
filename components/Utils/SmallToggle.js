// Thrid Party
import React from 'react';

const SmallToggle = ({ toggleFunc, toggleVal, names, className, disabled }) => {
  return (
    <div className={`flex text-sm rounded-sm overflow-hidden w-fit ${className} cursor-not-allowed text-primary `}>
      {names.map((name, index) => {
        return (
          <button
            disabled={disabled}
            key={index}
            onClick={() => toggleFunc(name)}
            className={`w-fit min-w-[80px] py-[5px] px-4 ${
              index ? 'rounded-r-sm' : 'rounded-l-sm'
            } border whitespace-nowrap ${
              toggleVal === name ? 'bg-secondary-button border-secondary-button' : ' border-web-gray'
            } ${disabled ? ' cursor-not-allowed' : ' cursor-pointer'}`}
          >
            {name}
          </button>
        );
      })}
    </div>
  );
};
export default SmallToggle;
