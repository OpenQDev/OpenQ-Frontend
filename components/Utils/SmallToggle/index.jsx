// Thrid Party
import React from 'react';

const SmallToggle = ({ toggleFunc, toggleVal, names, className }) => {
  return (
    <div className={`flex text-sm rounded-sm overflow-hidden w-fit ${className} text-primary `}>
      {names.map((name, index) => {
        return (
          <button
            key={index}
            onClick={() => toggleFunc(name)}
            className={`w-fit min-w-[80px] py-[5px] px-4 ${
              index ? 'rounded-r-sm' : 'rounded-l-sm'
            } border whitespace-nowrap ${
              toggleVal === name ? 'bg-secondary-button border-secondary-button' : ' border-web-gray'
            }`}
          >
            {name}
          </button>
        );
      })}
    </div>
  );
};
export default SmallToggle;
