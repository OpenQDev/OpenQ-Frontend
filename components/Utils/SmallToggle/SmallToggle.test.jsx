import React from 'react';
import SmallToggle from '.';
import renderer from 'react-test-renderer';

const names = ['Ready for work', 'All issues'];

describe('SmallToggle', () => {
  it('should atch DOM Snapshot', () => {
    const tree = renderer.create(<SmallToggle names={names} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
