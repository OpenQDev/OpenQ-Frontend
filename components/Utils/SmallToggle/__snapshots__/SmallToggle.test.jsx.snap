// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SmallToggle > should atch DOM Snapshot 1`] = `
<div
  className="flex text-sm rounded-sm overflow-hidden w-fit undefined text-primary "
>
  <button
    className="w-fit min-w-[80px] py-[5px] px-4 rounded-l-sm border whitespace-nowrap  border-web-gray"
    onClick={[Function]}
  >
    Ready for work
  </button>
  <button
    className="w-fit min-w-[80px] py-[5px] px-4 rounded-r-sm border whitespace-nowrap  border-web-gray"
    onClick={[Function]}
  >
    All issues
  </button>
</div>
`;
